cmd/otel-agent/subcommands/run/command.go

```diff
@@ -165,6 +167,9 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 		fx.Provide(func(cfg traceconfig.Component) telemetry.TelemetryCollector {
 			return telemetry.NewCollector(cfg.Object())
 		}),
+		fx.Provide(func(cfg traceconfig.Component) tracecompression.Component {
+			return gzip.NewComponent()
+		}),
```

Comment:
If I'm not mistake we should provide the fx version of each component. that way we could simply do 

```go
gzip "github.com/DataDog/datadog-agent/comp/trace/compression/fx-gzip/fx"
...


gzip.Module(),
...

```

You can see in the documentation and example of the different fx related folder 
https://datadoghq.dev/datadog-agent/components/creating-components/#the-fx-folders-compcompressionfx-zstdfxgo